app/views/applications/_applications_table.html.erb

```diff
@@ -23,8 +23,8 @@
         <td>
           <%= link_to application.name, application, class: 'js-open-on-submit' %>
         </td>
-        <td class="<%= application.staging_and_production_in_sync? ? "in-sync" : "not-in-sync" %>">
-          <%= application.staging_and_production_in_sync? ? "Yes" : "No" %>
+        <td class="<%= application.in_sync?(@environments) ? "in-sync" : "not-in-sync" %>">
```

Comment:
I'm not sure about changing the meaning of an established column without communicating it (I just signed into release and was shocked to see that almost nothing was in sync). 

I would either:

- create a new column to show if there are undeployed changes on integration
- use a different layout for "production and staging aren't in sync" (which could be bad and should be actioned so should be red), and "production/staging and integration aren't in sync" (which is not as bad so should not be red)

Reply:
> I'm not sure about changing the meaning of an established column without communicating it (I just signed into release and was shocked to see that almost nothing was in sync).

Yeah, sorry about the number of things needing releasing. One of the reasons I wanted to get this change out, was that I needed to release most apps due to Dependabot updates.

> I would either:
> 
>  -  create a new column to show if there are undeployed changes on integration
>  -  use a different layout for "production and staging aren't in sync" (which could be bad and should be actioned so should be red), and "production/staging and integration aren't in sync" (which is not as bad so should not be red)

I'm not convinced that production and staging aren't in sync is always actionable. It's reasonable in my view to deploy to staging, and then leave a change there overnight, as that's a useful testing technique given the traffic replay. I'm not sure about adding more columns, but I'll have another think about the table.

Reply:
> Yeah, sorry about the number of things needing releasing. One of the reasons I wanted to get this change out, was that I needed to release most apps due to Dependabot updates.

I have no issue with how it was released - the problem is that for years red in the column meant a staging-prod difference, but now it means something else. The UI doesn't give a clue that this has changed, so this is quite confusing.

> I'm not convinced that production and staging aren't in sync is always actionable. It's reasonable in my view to deploy to staging, and then leave a change there overnight, as that's a useful testing technique given the traffic replay. I'm not sure about adding more columns, but I'll have another think about the table.

Maybe actionable wasn't the right word. The key thing is that if staging and production aren't in sync, there's something going on and you need to pay attention. If integration isn't deployed that's just very normal.

Reply:
> I have no issue with how it was released - the problem is that for years red in the column meant a staging-prod difference, but now it means something else. The UI doesn't give a clue that this has changed, so this is quite confusing.

Ok, my thinking has been that "In sync?" could be seen to represent the state of the environments shown, but I get that changing it is confusing. I've now got a Pull Request up [1] that tries to make the status of applications clearer, and restores red to be used to indicate production/staging differences.

1: https://github.com/alphagov/release/pull/253